### DISCUSSION TITLE
Looking for file or path to database in prometheus where alerts are stored in json or other format that reflects in grafana

### DISCUSSION THREAD

**Discussion by MuhammadMunir12** [prometheus/prometheus#10364] [Q&A]
**What did you do?**
I am not sure whether this is right forum to ask this questions but I am looking for file or path to database in prometheus where alerts are stored in json or other format that reflects in grafana. I want to write a scripts to take some actions based on alerts/triggers. Normally alerts can be seen on dashboard and notified accordingly on email or messages. My concerns is to trace the path from where the alerts are relflected in grafana. 

**What did you expect to see?**
I expect to have path to database or any json file in which alerts are populated on run time.

**Environment**

* System information:

	Linux 3.10.0-1160.53.1.el7.x86_64 x86_64

* Prometheus version:

	prometheus, version 2.30.0-rc.0 (branch: HEAD, revision: 05a816bfb739b3841acd82bd285bfb5dfec7bfd7)
        build user:       root@438506e6c112
        build date:       20210909-13:31:07
        go version:       go1.17
        platform:         linux/amd64

* Alertmanager version:

	alertmanager, version 0.20.0 (branch: HEAD, revision: f74be0400a6243d10bb53812d6fa408ad71ff32d)
        build user:       root@00c3106655f8
        build date:       20191211-14:13:14
        go version:       go1.13.5

Any help in this regard would be highly appreciated.

**Comment by roidelapluie** [comment]
Prometheus populates alerts in its TSDB, you can query the `ALERTS` metric.

**Reply by roidelapluie** [reply]
You could also query the /api/v1/alerts endpoint.
